CBinding/Parser/Symbol.cs

```diff
@@ -0,0 +1,204 @@
+using System;
+using ClangSharp;
+using System.Collections.Generic;
+using GLib;
+using System.Runtime.InteropServices;
+using System.IO;
+using Gdk;
+
+namespace CBinding.Parser
+{
+	/// <summary>
+	/// Represents clang abstract symbols from the built AST
+	/// </summary>
+	public class Symbol
+	{
+		protected CProject project;
```

Comment:
Instead of protected backing fields why not convert the properties to use auto properties with a protected set?